packages/wrangler/src/config/config-helpers.ts

```diff
@@ -1,35 +1,139 @@
-import path from "path";
+import fs from "node:fs";
+import path from "node:path";
 import { findUpSync } from "find-up";
+import dedent from "ts-dedent";
+import { UserError } from "../errors";
+import { logger } from "../logger";
+import { formatMessage, ParseError, parseJSONC, readFileSync } from "../parse";
+
+export type ResolveConfigPathOptions = {
+	useRedirectIfAvailable?: boolean;
+};
+
+export type ConfigPaths = {
+	/** Absolute path to the actual configuration being used (possibly redirected from the user's config). */
+	configPath: string | undefined;
+	/** Absolute path to the user's configuration, which may not be the same as `configPath` if it was redirected. */
+	userConfigPath: string | undefined;
+};
 
 /**
  * Resolve the path to the configuration file, given the `config` and `script` optional command line arguments.
  * `config` takes precedence, then `script`, then we just use the cwd.
+ *
+ * Returns an object with two paths: `configPath` and `userConfigPath`. If defined these are absolute file paths.
  */
-export function resolveWranglerConfigPath({
-	config,
-	script,
-}: {
-	config?: string;
-	script?: string;
-}): string | undefined {
+export function resolveWranglerConfigPath(
+	{
+		config,
+		script,
+	}: {
+		config?: string;
+		script?: string;
+	},
+	options: { useRedirectIfAvailable?: boolean }
```

Comment:
Should this (and the default value/behavior) be documented?

Reply:
Yes documentation is in https://github.com/cloudflare/cloudflare-docs/pull/18757